File: benchmark/comparison/tasks/19/solution.py
import math

def shape_area(shape, *args):
    if shape == "Circle":
        r = args[0]
        return 3.14159 * r * r
    elif shape == "Rect":
        w, h = args[0], args[1]
        return float(w * h)
    elif shape == "Triangle":
        b, h = args[0], args[1]
        return float(b * h / 2)

print(shape_area("Circle", 5))
print(shape_area("Rect", 3, 4))
print(shape_area("Triangle", 6, 4))

File: benchmark/comparison/tasks/19/solution.ts
type Shape =
  | { tag: "Circle"; radius: number }
  | { tag: "Rect"; width: number; height: number }
  | { tag: "Triangle"; base: number; height: number };

function area(shape: Shape): number {
  switch (shape.tag) {
    case "Circle":
      return 3.14159 * shape.radius * shape.radius;
    case "Rect":
      return shape.width * shape.height;
    case "Triangle":
      return 0.5 * shape.base * shape.height;
  }
}

function formatArea(val: number): string {
  if (Number.isInteger(val)) {
    return val.toFixed(1);
  }
  // Remove trailing zeros but keep at least one decimal
  const s = val.toString();
  return s;
}

console.log(formatArea(area({ tag: "Circle", radius: 5 })));
console.log(formatArea(area({ tag: "Rect", width: 3, height: 4 })));
console.log(formatArea(area({ tag: "Triangle", base: 6, height: 4 })));
